### PR TITLE
Fix PL2Bat (ERRORLEVEL and environment handling), tests, and AppVeyor CI

### PR DESCRIPTION
**Pull request by rivy**
PR includes multiple fixes (further detailed in the commits):

- PL2Bat fixes for "echo", "ERRORLEVEL", and environment state handling
- Testing includes process exit status as well as expanded exit value testing
- AppVeyor now correctly tests for perl versions back to v5.8.9

### CODE REVIEW

File: t/make_executable.t [hunk 1]
```diff
@@ -2,32 +2,50 @@
...
 use warnings FATAL => 'all';
+use English;
 
 use Config;
 use Test::More;
 use ExtUtils::PL2Bat;
 use Cwd qw/cwd/;
 
-plan($^O eq 'MSWin32' ? (tests => 7) : skip_all => 'Only usable on Windows');
+use IPC::System::Simple qw/ capture system EXIT_ANY /;
```
Comment (Leont) [inline t/make_executable.t]: As this distribution is intended to be included in core, non-core dependencies are not appropraite.

**Comment by rivy** [comment]
Ok.

I'm happy to change it... suggestions on a core alternative?

**Comment by rivy** [comment]
I refactored the tests to use only `qx/.../` command execution. I added some minor CMD shell hackery to enable testing the ERRORLEVEL from `qx` (which otherwise mangles the ERRORLEVEL/exit code down into a 7 bit return value in $CHILD_ERROR).

Let me know if that is an improvement and/or needs more modifications.

**Comment by rivy** [comment]
@Leont , I fixed another couple of typos.

Is the current version now acceptable?

**Comment by Leont** [comment]
> Is the current version now acceptable?

I'll discuss it on the [Perl Toolchain Summit](http://act.qa-hackathon.org/pts2018/)

**Comment by rivy** [comment]
@Leont ... just checking back in; it's been over a year. In the meanwhile, I've added CI testing for windows and linux (via AppVeyor and Travis), both successfully testing back to perl v5.8.

Let me know if you need something else to complete this PR.

**Comment by Leont** [comment]
I primarily need someone with more windows knowledge than me to do some reviewing. @haarg?

**Comment by haarg** [comment]
`setlocal` requires `cmd.exe` and thus Windows NT.  It doesn't make sense to use it as well as trying to have other handling for `command.com`.  If we want to drop support for `command.com`, the entire script could be substantially simplified.

**Comment by rivy** [comment]
Hmm, I believe I could make it work without `setlocal`. `%ERRORLEVEL%` would be clobbered by the script, but that seems reasonable. Otherwise, I'd just need to remove the `@echo off` line and add a lot of `@` prefixes to prevent echo'd lines.

**Comment by rivy** [comment]
And, @haarg , thanks for the code review fly-by!

**Comment by rivy** [comment]
Yes, I can eliminate `setlocal` and `echo off`.

I added a test to assure that there's no extraneous output and then refactored the BAT wrap without `setlocal` or `echo off`, using only leading `@` to suppress output (see <https://github.com/rivy/perl.ExtUtils-PL2Bat/commit/1a120d5562e5e9f6ccaa6796d241d1dc290d72d3> with AppVeyor testing at <https://ci.appveyor.com/project/rivy/perl-extutils-pl2bat/builds/22471711>).

If that seems like a better idea, I'll clean up the commits/messages and push it back onto this feature branch.

**Comment by Leont** [comment]
> It doesn't make sense to use it as well as trying to have other handling for `command.com`. If we want to drop support for `command.com`, the entire script could be substantially simplified.

That sounds like a good idea. Though right now the "priority" is to get all three other forks of this codebase to use this thing first, so that improvements are shared.

**Comment by Leont** [comment]
Why add the travis configuration if the tests will be skipped on non-windows?

**Comment by rivy** [comment]
Good question.
I believe I put it in for completeness to make sure the build/test functionality works correctly under *nix, especially since `dzil` is not usable or testable on the AppVeyor Windows side. And, in looking at the Travis CI report, `dzil` the "t/00-compile.t" test passes (brought in by travis helpers?).

But it obviously doesn't need to test multiple versions. That's just an artifact of me putting my Travis CI config template in, instead of customizing it for this module. I should at least just cut it down to one or two versions, maybe just "stable".

Ultimately, if it's not useful information, I can cut that one commit back out.

**Comment by rivy** [comment]
@Leont , what do I need to do to get this over the finish line? It's been on my PR list for over a year now.

**Comment by Leont** [comment]
> what do I need to do to get this over the finish line? It's been on my PR list for over a year now.

I'm planning to take a last look at it at the [Perl Toolchain Summit](https://act.qa-hackathon.org/pts2019/), expecting to merge it then.

In the mean time, cleaning up the history a bit would make my life more pleasant here.

**Comment by rivy** [comment]
# 🧹
* done

**Pull request merged**